File: tests/adapters/c/goldens/imports/strip_all.c
/**
 * C module for testing include optimization.
 */

// Standard library includes (external)
// … 12 imports omitted

// POSIX headers (external/system)
// … 8 imports omitted

// Common external libraries (external)
// … 8 imports omitted

// Local/project includes (should be considered local)
// … 9 imports omitted

// Relative includes with different depth levels
// … 3 imports omitted

// Long include lists from single directory (candidates for summarization)
// … 23 imports omitted

typedef struct {
    void* user_service;
    void* db_connection;
    void* logger;
} ImportTestService;

ImportTestService* import_test_service_new(
    void* user_service,
    void* db_connection,
    void* logger
) {
    ImportTestService* service = (ImportTestService*)malloc(sizeof(ImportTestService));
    if (!service) return NULL;

    service->user_service = user_service;
    service->db_connection = db_connection;
    service->logger = logger;

    return service;
}

void* process_data(ImportTestService* service, void** data, int count) {
    if (!service || !data) return NULL;

    // Using external libraries (zlib for compression)
    z_stream stream;
    memset(&stream, 0, sizeof(stream));

    // Using standard library functions
    time_t now = time(NULL);
    struct tm* timeinfo = localtime(&now);

    // Using POSIX functions
    pid_t pid = getpid();

    // Using local utilities
    // (would call validation functions here)

    return NULL;
}

int make_http_request(ImportTestService* service, const char* url) {
    if (!service || !url) return -1;

    // Using libcurl
    CURL* curl = curl_easy_init();
    if (!curl) {
        fprintf(stderr, "Failed to initialize curl\n");
        return -1;
    }

    curl_easy_setopt(curl, CURLOPT_URL, url);
    curl_easy_setopt(curl, CURLOPT_TIMEOUT, 5L);
    curl_easy_setopt(curl, CURLOPT_USERAGENT, "ImportTestService/1.0");

    CURLcode res = curl_easy_perform(curl);
    curl_easy_cleanup(curl);

    if (res != CURLE_OK) {
        fprintf(stderr, "HTTP request failed: %s\n", curl_easy_strerror(res));
        return -1;
    }

    return 0;
}

char* serialize_data(void* data) {
    if (!data) return NULL;

    // Using json-c
    json_object* jobj = json_object_new_object();
    if (!jobj) return NULL;

    // Add data to JSON object
    json_object_object_add(jobj, "data", json_object_new_string("test"));

    const char* json_str = json_object_to_json_string(jobj);
    char* result = strdup(json_str);

    json_object_put(jobj);
    return result;
}

int query_database(ImportTestService* service, const char* sql) {
    if (!service || !sql) return -1;

    // Using libpq (PostgreSQL)
    PGconn* conn = NULL; // Would connect here
    if (!conn) {
        fprintf(stderr, "Database connection failed\n");
        return -1;
    }

    PGresult* res = PQexec(conn, sql);
    if (PQresultStatus(res) != PGRES_TUPLES_OK) {
        fprintf(stderr, "Query failed: %s\n", PQerrorMessage(conn));
        PQclear(res);
        return -1;
    }

    int rows = PQntuples(res);
    PQclear(res);

    return rows;
}

void free_import_test_service(ImportTestService* service) {
    if (!service) return;
    free(service);
}

File: tests/adapters/rust/goldens/imports/strip_all.rs
// Rust module for testing import optimization.

// … 91 imports omitted (72 lines)

struct ImportTestService {
    user_service: UserService,
    db_connection: Connection,
    logger: Box<dyn std::any::Any>,
}

impl ImportTestService {
    fn new(user_service: UserService, db_connection: Connection, logger: Box<dyn std::any::Any>) -> Self {
        Self {
            user_service,
            db_connection,
            logger,
        }
    }

    fn process_data(&self, data: Vec<Box<dyn std::any::Any>>) -> Vec<HashMap<String, String>> {
        // Using standard library
        let mut processed = Vec::new();

        for item in data {
            let mut result = HashMap::new();
            // Using chrono for timestamps
            let timestamp = Utc::now().to_rfc3339();
            result.insert("timestamp".to_string(), timestamp);
            processed.push(result);
        }

        // Using local utilities (would call validation functions here)
        processed
    }

    async fn make_http_request(&self, url: &str) -> Result<String> {
        // Using reqwest
        let client = Client::builder()
            .timeout(Duration::from_secs(5))
            .user_agent("ImportTestService/1.0")
            .build()?;

        let response = client.get(url).send().await?;

        if !response.status().is_success() {
            anyhow::bail!("HTTP request failed with status: {}", response.status());
        }

        let text = response.text().await?;
        Ok(text)
    }

    fn serialize_data(&self, data: &dyn std::any::Any) -> Result<String> {
        // Using serde_json
        let json_value = json!({
            "data": "test"
        });

        Ok(serde_json::to_string_pretty(&json_value)?)
    }

    async fn query_database(&self, sql: &str) -> Result<Vec<Row>> {
        // Using sqlx
        let pool = PgPool::connect("postgresql://localhost/test").await?;

        let rows = sqlx::query(sql).fetch_all(&pool).await?;

        Ok(rows)
    }

    async fn start_server(&self) -> Result<()> {
        // Using actix-web
        HttpServer::new(|| {
            App::new()
                .route("/health", web::get().to(|| async { HttpResponse::Ok().body("OK") }))
        })
        .bind(("127.0.0.1", 8080))?
        .run()
        .await?;

        Ok(())
    }
}

// Forward declarations (should not be treated as imports)
struct Forward;
mod module_forward;
